### PR TITLE
adding expect and documentation for Str builtin

### PR DESCRIPTION
**Pull request by lukewilliamboswell**
I have found it difficult to decipher some of the builtin functionality. I found the docs that had an expect call to be very helpful, so I've started adding them and checking they compile with ```roc test```. I'm not removing any content, though I have been trying to align the docs and make them more consistently formatted. Where there is no note I have been trying to add a description of the function.

Still a WIP.

### CODE REVIEW

**Comment by lukewilliamboswell** [comment]
I completed my edits for the Str builtin. I also made some minor edits in Num builtin. In the end I decided to stick with the ```##     expect ...``` format with 5 spaces as that enables type checking to work correctly. I tried to ensure all the types were formatted with [Str] and all functions were formatted \`my function\`. 

This should be ready for review. My intention is to create another PR if I do another builtin.

File: crates/compiler/builtins/roc/Str.roc [hunk 1]
```diff
@@ -140,40 +140,44 @@ Utf8Problem : { byteIndex : Nat, problem : Utf8ByteProblem }
...
-## >>> Str.isEmpty "hi!"
-##
-## >>> Str.isEmpty ""
+##     expect Str.isEmpty "hi!" == Bool.false
+##     expect Str.isEmpty "" == Bool.true
 isEmpty : Str -> Bool
+
+## Concatenate two [Str] values together. 
+##
+## >>> expect Str.concat "Hello" "World" == "HelloWorld"
```
Comment (bhansconnect) [inline crates/compiler/builtins/roc/Str.roc:149]: The use of >>> seems inconsistent.

Comment (lukewilliamboswell) [reply crates/compiler/builtins/roc/Str.roc]: Thanks, I missed some. Should be all gone now.

**Comment by lukewilliamboswell** [comment]
It looks like there is a new Roc.box builtin which ```roc format``` doesn't like.

**Review by Anton-4 — APPROVED** [review]
This will be valuable to a lot of people, thanks @lukewilliamboswell!

**Comment by lukewilliamboswell** [comment]
Looks the CI is complaining about an issue in ```Dict.roc``` with a missing implementation for ```listGetUnsafe : List a, Nat -> a```

**Review by Anton-4 — APPROVED** [review]

**Pull request merged**